Source/FieldSolver/ElectrostaticSolver.cpp

```diff
@@ -778,10 +778,13 @@ void ElectrostaticSolver::BoundaryHandler::definePhiBCs ( )
                 dirichlet_flag[idim*2] = false;
             }
             else {
-                AMREX_ALWAYS_ASSERT_WITH_MESSAGE(false,
-                    "Field boundary conditions have to be either periodic, PEC or none "
-                    "when using the electrostatic solver"
+                WarpX::GetInstance().RecordWarning(
```

Comment:
Perhaps add a specific check for PML boundaries here, and print the warning then. This should still probably do the assert to check for bad BC's in case there's a typo for instance or other bad name.